package/MDAnalysis/analysis/pca.py

```diff
@@ -265,12 +269,25 @@ def _conclude(self):
         self.cov /= self.n_frames - 1
         e_vals, e_vects = np.linalg.eig(self.cov)
         sort_idx = np.argsort(e_vals)[::-1]
-        self.variance = e_vals[sort_idx]
-        self.variance = self.variance[:self.n_components]
-        self.p_components = e_vects[:self.n_components, sort_idx]
-        self.cumulated_variance = (np.cumsum(self.variance) /
-                                   np.sum(self.variance))
+        self._variance = e_vals[sort_idx]
+        self._p_components = e_vects[:, sort_idx]
         self._calculated = True
+        self.n_components = self._n_components
+
+    @property
+    def n_components(self):
+        return self._n_components
+
+    @n_components.setter
+    def n_components(self, n):
+        if self._calculated:
+            if n is None:
+                n = len(self._variance)
+            self.variance = self._variance[:n]
```

Comment:
Do we really want to permanently change `self.variance`?

In a recent discussion with @ianmkenney and @VOD555 we found it confusing that the "total" variance might only be  calculated from a subset of components. The docs were not very clear on this to start with. This can be confusing if you want to plot how much each PC contributes to the total observed variance but only selected, say,  `n=5`.

Reply:
No, hence the addition of the new attributes `_variance` and `_p_components`. The cumulative variance is also calculated from the total `_variance` so that now it does not add up to 1 if `n_components` is not `None`. TBH I don't really get the idea of throwing away information that was expensive to compute and is *already in memory*, so setting `n_components` is more of a convenient way to truncate or extend the visible variance, cumulated_variance, and p_components. It does not alter their values. We could suggest to users that they manually set ``self._variance = self.variance`` if they want to save space.

Reply:
This should fix the issue that  @ianmkenney and @VOD555 saw – guys, if you still see a problem here, please say something now.